resolver/GlobalPass.cc

```diff
@@ -258,17 +258,16 @@ void resolveTypeMembers(core::GlobalState &gs, core::ClassOrModuleRef sym,
             auto attachedClass = singleton.data(gs)->findMember(gs, core::Names::Constants::AttachedClass());
             if (attachedClass.exists()) {
                 auto attachedClassTypeMember = attachedClass.asTypeMemberRef().data(gs);
-                if (singleton.data(gs)->flags.isFinal) {
+                bool isFinal = singleton.data(gs)->flags.isFinal;
+                if (isFinal) {
```

Comment:
TODO(jez) This commit is sufficient to fix it, but we should probably look into the original `isFixed` approach:

- test how disruptive it is to no longer be able to apply type arguments to `T.class_of` in a final class
- adjust the types printing code, and any other code that deals with T.attached_class and depends on calling `typeArity` as a cheap way to check something.